pkg/3scale/amp/operator/base_apimanager_logic_reconciler.go

```diff
@@ -85,14 +87,18 @@ func (r *BaseAPIManagerLogicReconciler) ReconcileRoleBinding(desired *rbacv1.Rol
 }
 
 func (r *BaseAPIManagerLogicReconciler) ReconcileGrafanaDashboard(desired *grafanav1alpha1.GrafanaDashboard, mutateFn reconcilers.MutateFn) error {
+	if !r.apiManager.IsMonitoringEnabled() {
+		return nil
+	}
+
 	kindExists, err := r.HasGrafanaDashboards()
 	if err != nil {
 		return err
 	}
-
 	if !kindExists {
```

Comment:
Before we always logged when the kind didn't exist, independently on whether monitoring was enabled or not, and we logged the message as Info type.
Now we only log that message when monitoring is enabled and the kind doesn't exist, and in that case we return as an error. However, I'm not entirely sure about this last behavior. If monitoring is enabled and the kind doesn't exist, should we stop processing the next resources?